hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -50,7 +50,7 @@
   private boolean closed;
 
   public HiveCatalog(Configuration conf) {
-    this.clients = new HiveClientPool(2, conf);
+    this.clients = new HiveClientPool(conf);
```

Comment:
This will bump the default size to 5. Maybe, we can change the default value to 2.